docs/background-tasks-with-celery.rst

```diff
@@ -0,0 +1,66 @@
+Background tasks with Celery
+============================
+
+Sometimes, a user on Ambuda will request an operation that takes a long time
+for our application code to process. Some examples of long-running tasks include:
+
+- calling a third-party API like Google OCR.
+- splitting a large PDF into separate images.
+- sending an email through a third-party mail server.
+- running a regular batch job to convert published texts into PDFs.
+
+To handle these kinds of background tasks, we use the `Celery`_, a Python task
+scheduler that runs in a separate process from our main application code.
+
+.. _Celery: https://docs.celeryq.dev/en/stable/
+
+
+Quickstart
+----------
+
+First, install Redis, which we use to communicate with the Celery process::
+
+    # Sorry, OSX only for now.
+    # If you're on a Linux system, send us a PR!
+    ./scripts/install_osx_dependencies.sh
+
+Then, run Celery in a new terminal window::
+
+    make celery
+
+
+Basic mechanics of Celery
+-------------------------
+
+The basic mechanics of Celery are as follows:
+
+1. On our server, we have two processes: our web application code and a Celery
+   process that manages a set of worker processes.
+
+2. When we need to create a long-running request, our application sends a
+   non-blocking request to Celery. We send this message through a *message
+   broker* such as Redis.
+
+3. Celery receives the request and assigns it to one of its workers.
+
+4. The worker can optionally send status updates to a *backend* component such
+   as Redis.
```

Comment:
Is Redis both the message broker and backend component?

Reply:
Yes -- will update the docs with our setup